enterprise/frontend/src/embedding-sdk/cli/utils/get-component-snippets.ts

```diff
@@ -26,33 +35,53 @@ export function getComponentSnippets(options: Options) {
     ? ANALYTICS_PROVIDER_SNIPPET_WITH_USER_SWITCHER
     : ANALYTICS_PROVIDER_SNIPPET_MINIMAL;
 
-  const components = [
+  const components: SampleComponent[] = [
     {
-      name: "analytics-provider",
-      content: analyticsProviderSnippet.trim(),
+      fileName: "analytics-provider",
+      componentName: "AnalyticsProvider",
+      content: withNextJsUseClientDirective(
+        analyticsProviderSnippet,
+        isNextJs,
+      ).trim(),
```

Comment:
Next.js requires any component that uses `useContext`, `useState` or `useReducer` to have `"use client"`. In this case, we need to store the state for the theme switcher (light vs dark mode) at the example provider.